R/tlf_ae_specific.R

```diff
@@ -67,57 +68,182 @@ tlf_ae_specific <- function(outdata,
+    )
+  }
+
   footnotes <- vapply(footnotes, glue::glue_data,
     .x = list(medra_version = medra_version), FUN.VALUE = character(1)
   )
   names(footnotes) <- NULL
 
-  # Define column header
-  colheader_n <- c(
-    paste0(" | ", paste(group, collapse = " | ")),
-    paste0(" | ", paste(rep("n | (%)", n_group), collapse = " | "))
+  # within (group statistics)
+
+  col_tbl_within <- strsplit(names(tbl), "_") |>
+    unlist() |>
+    (\(list) list[list %in% c("n", "prop", "dur", "events")])() |>
+    unique()
+
+  colhead_within <- paste(
+    sapply(
+      X = col_tbl_within,
+      FUN = switch,
+      "n" = "n",
+      "prop" = "(%)",
+      "dur" = "Mean Duration (SE)",
+      "events" = "Mean Events per Participant (SE)"
+    ),
+    collapse = " | "
+  )
+
+  colhead_1_within <- paste(group, collapse = " | ")
+
+  colhead_2_within <- paste(rep(colhead_within, n_group),
+    collapse = " | "
+  )
+
+  colborder_within <- sapply(
+    X = col_tbl_within,
+    FUN = switch,
+    "n" = "single",
+    "prop" = "",
+    "dur" = "single",
+    "events" = "single",
+    USE.NAMES = FALSE
+  )
+
+  rwidth_2_within <- rep(1, length(col_tbl_within) * n_group)
+
+  rwidth_1_within <- tapply(
+    rwidth_2_within,
+    c(rep(1:n_group, each = length(col_tbl_within))),
+    sum
   )
 
-  # TODO: add logic for CI and p-value with multipel groups following WMA mock up table.
-  # colheader_ci <- c(paste("Difference in % vs", group[reference_group]),
-  # "Estimate (95% CI)")
+  colborder_within <- rep(colborder_within, n_group)
 
-  # colheader_p <- c("", "p-value")
-  # colheader <- paste(colheader_n, colheader_ci, colheader_p, sep = " | ")
+  # between (comparison statistics)
 
-  colheader <- colheader_n
+  col_tbl_between <- strsplit(names(tbl), "_") |>
+    unlist() |>
+    (\(list) list[list %in% c("diff", "ci", "p")])() |>
+    unique()
+
+  if (length(col_tbl_between) > 0) {
+    colhead_between <- paste(
+      sapply(
+        X = col_tbl_between,
+        FUN = switch,
+        "diff" = "Estimate",
+        "ci" = paste0("(", outdata$ci_level * 100, "% CI)"),
+        "p" = "p-value",
+      ),
+      collapse = " | "
+    )
+
+    colhead_1_between <- paste("Difference in %",
+      outdata$group[between_total[-reference_group]],
+      "vs.",
+      outdata$group[reference],
+      collapse = " | "
+    )
+
+    colhead_2_between <- paste(rep(colhead_between, n_comparisons), collapse = " | ")
+
+    colborder_between <- sapply(
+      X = col_tbl_between,
+      FUN = switch,
+      "diff" = "single",
+      "ci" = "",
+      "p" = "single",
+      USE.NAMES = FALSE
+    )
+
+    rwidth_2_between <- rep(1, length(col_tbl_between) * n_comparisons)
+
+    rwidth_1_between <- tapply(
+      rwidth_2_between,
+      c(rep(1:n_comparisons, each = length(col_tbl_between))),
+      sum
+    )
+
+    colborder_between <- rep(colborder_between, n_comparisons)
+  } else {
+    colhead_between <- colhead_1_between <- colhead_2_between <- NULL
```

Comment:
if there are no `between` objects, set them all to null to allow for clean `paste()`-ing and `c()`ing with `within` objects